smithy-model/src/test/resources/software/amazon/smithy/model/errorfiles/validators/paginated/paginated-map-tokens.errors

```diff
@@ -1,2 +1,2 @@
-[DANGER] ns.foo#MapTokens: paginated trait `inputToken` member `MapTokensInput` targets a map shape, but this is not recommended. One of [`string`] SHOULD be targeted. | PaginatedTrait
-[DANGER] ns.foo#MapTokens: paginated trait `outputToken` member `MapTokensOutput` targets a map shape, but this is not recommended. One of [`string`] SHOULD be targeted. | PaginatedTrait
+[DANGER] ns.foo#MapTokens: paginated trait `inputToken` member `MapTokensInput` targets a map shape, but this is not recommended. One of [`string`] SHOULD be targeted. | PaginatedTrait.inputToken.MapTokensInput.WRONG_SHAPE_TYPE.map
```

Comment:
The input shape and the all the additional information beyond the fact that it's inputToken can be removed. They don't add to the uniqueness of the event.

`PaginatedTrait.WrongShapeType.inputToken`

This matches the format of the others, and you don't really need to know any more information.

Reply:
> Is there not a case where one trait can have multiple members with the wrong target type?

The validator looks for a very specific set of members based on the trait. So if there are multiple issues, then there would be different events: `PaginatedTrait.WrongShapeType.inputToken`, `PaginatedTrait.WrongShapeType.outputToken`, `PaginatedTrait.WrongShapeType.pageSize`, etc.

> Also, "map" needs to be included as part of the eventId so that if they change it to target a different but also wrong shape, we will get a new non-deduped eventId

I don't think so. Event IDs don't have to be so unique that they account for breaking changes like changing the target shape of a paginator. We have bigger problems for changes like that, and we should err towards simplicity for these.

Reply:
> Event IDs don't have to be so unique that they account for breaking changes like changing the target shape of a paginator. 

Well sure the semantic diff or other would catch it, but Tony and I were thinking that it would still be good to reraise the violation in that case so that they _do_ know they are doing something they shouldn't. Maybe it's a case where they are intentionally changing the target shape and have escalated to get approval to break backwards compatibility, but don't even know that they are (still) doing something not recommended.

> The validator looks for a very specific set of members based on the trait. So if there are multiple issues, then there would be different events: PaginatedTrait.WrongShapeType.inputToken, PaginatedTrait.WrongShapeType.outputToken, PaginatedTrait.WrongShapeType.pageSize, etc.

What about within just one of those, such as the inputToken? Can inputToken not have multiple members that all target wrong shape types?

Reply:
I think we need to err towards simplicity here and in all events. There will be other Smithy implementations in the future, and they're going to have to reimplement all of this.

> What about within just one of those, such as the inputToken? Can inputToken not have multiple members that all target wrong shape types?

If I understand what you're asking, then no. The event is bound to the operation (ns.foo#MapTokens), and there is a single inputToken binding per/operation.

Reply:
ok, sounds fine to me